xarray/core/common.py

```diff
@@ -748,244 +751,12 @@ def pipe(
         else:
             return func(self, *args, **kwargs)
 
-    def groupby(
```

Comment:
Is it not possible to preserve this?

Reply:
I assume that it could be possible with lots of generics, but I think the change is cleaner and easier to understand.